### PR TITLE
Smart contract page

### PR DESCRIPTION
**Pull request by BruceJi7**
Add Cypress tests for Smart Contracts pages

### CODE REVIEW

File: cypress/integration/test_tables.js [hunk 1]
```diff
@@ -1,136 +1,121 @@
...
+      const value = $firstBlockHash.text()
 
+      cy.wait(500)
 
-        cy.get('#pagination-previous').click()
-        cy.get('#pagination-previous').click()
-        cy.get('#pagination-previous').click()
-        cy.get('#pagination-previous').click()
-        cy.get('#pagination-previous').click()
+      cy.get("#pagination-next").click()
```
Comment (ivanms1) [inline cypress/integration/test_tables.js:44]: is there a way to make this more dry?
do for loops work here?

File: src/components/SmartContracts/ContractDetails/ContractCodeTab/ContractCodeTab.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const ContractCodeTab = () => {
+  return <div className="contract-code monofont">CODE HERE</div>
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractDetails/ContractCodeTab/ContractCodeTab.tsx:2]: Waiting on backend?

Comment (BruceJi7) [reply src/components/SmartContracts/ContractDetails/ContractCodeTab/ContractCodeTab.tsx:2]: Yes

File: cypress/integration/test_tables.js (hunk 1 above)
Comment (BruceJi7) [reply cypress/integration/test_tables.js:44]: I didn't really see something about how to do that with a loop. Rather, you can make it repeat the whole test over and over

File: src/components/SmartContracts/ContractDetails/ContractTextBlock/ContractTextBlock.tsx [hunk 3]
```diff
@@ -1,26 +1,17 @@
...
-          nihil impedit odit suscipit ex dolorum! Laudantium quaerat dicta,
-          tempore deserunt qui labore. Lorem ipsum dolor sit amet consectetur
-          adipisicing elit. Et autem doloribus consectetur? Dolore delectus,
-          repudiandae vitae magnam, quasi autem corporis neque atque, ratione
-          quas error. Alias saepe rerum ad ab.
-        </div>
-      ) : (
-        <div className="contract-description-tab monofont">CODE TIME FOOLS</div>
-      )}
+      <div className="contract-description">{content}</div>
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractDetails/ContractTextBlock/ContractTextBlock.tsx:14]: make into component

File: src/components/SmartContracts/ContractDetails/TabMenu/TabMenu.tsx [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
-        }}
-      >
-        Code
-      </div>
+      {tabs.map((t) => {
+        return (
+          <div
+            key={t}
+            className={activeTab === t ? "tab active-tab" : "tab"}
+            onClick={() => {
```
Comment (ivanms1) [inline src/components/SmartContracts/ContractDetails/TabMenu/TabMenu.tsx:15]:
```suggestion
            onClick={() => doChangeTab(t)}
```

File: src/utility/functions.ts [hunk 5]
```diff
@@ -54,3 +54,25 @@ export const logger = (message: string, level: string, ...data: any[]) => {
...
+    case "conToken":
+    case "conun":
+      return "coin"
+
+    case "ConunDrive":
+    case "drive_1":
+    case "drive":
+      return "drive"
+
+    case "_lifecycle":
```
Comment (ivanms1) [inline src/utility/functions.ts:72]: remove pls

**Pull request merged**